### PR TITLE
Add convolution example to README.md

### PR DESCRIPTION
**Pull request by alex-s-gardner**
Thought this would be useful for other as it took me a bit to figure out

### CODE REVIEW

**Review by rafaqz — COMMENTED** [review]
Cool thanks! Sorry there is so little help with that. Could it also go in the `Kernel` docs?

Maybe note it could also use the `Base.dot` product, but `kernelproduct` lets you use e.g an array of StaticArray and it will still work (`dot` is recursive).

We should also show a split gausian convolution optimisaption with SwitchingStencilArray and Horizontal/Vertical passes some day, and see how much it beats ImageFiltering.jl by ;)

**Comment by alex-s-gardner** [comment]
Just an FYI, applying a 5x5 kernel to a 2931 by 7217 array I get a 15X speedup over imfilter!

**Comment by rafaqz** [comment]
Try it on your GPU now

File: src/stencils/kernel.jl [hunk 1]
```diff
@@ -61,6 +61,33 @@ the stencil and kernel, like [`kernelproduct`](@ref) can be used in
...
+# Define a stencil that is the same size as the kernel array
+stencil = Window(1)
+
+# Create a stencil Kernel from the stencil and the kernel array
+k = Kernel(stencil, sharpen)
+
+# Wrap the random array and the Kernel in a StencilArray
+A = StencilArray(r, k)
+
+# use `mapstencil` with the `kernelproduct` funciton to convolve the Kernel with array
```
Comment (rafaqz) [inline src/stencils/kernel.jl:86]:
```suggestion
# use `mapstencil` with the `kernelproduct` function to convolve the Kernel with array
```

File: README.md [hunk 2]
```diff
@@ -202,6 +202,35 @@ CUDA.@time mapstencil(mean, A);
...
+# Define a stencil that is the same size as the kernel array
+stencil = Window(1)
+
+# Create a stencil Kernel from the stencil and the kernel array
+k = Kernel(stencil, sharpen)
+
+# Wrap the random array and the Kernel in a StencilArray
+A = StencilArray(r, k)
+
+# use `mapstencil` with the `kernelproduct` funciton to convolve the Kernel with array.
```
Comment (rafaqz) [inline README.md:228]:
```suggestion
# use `mapstencil` with the `kernelproduct` function to convolve the Kernel with array. 
```

File: README.md [hunk 3]
```diff
@@ -202,6 +202,35 @@ CUDA.@time mapstencil(mean, A);
...
+stencil = Window(1)
+
+# Create a stencil Kernel from the stencil and the kernel array
+k = Kernel(stencil, sharpen)
+
+# Wrap the random array and the Kernel in a StencilArray
+A = StencilArray(r, k)
+
+# use `mapstencil` with the `kernelproduct` function to convolve the Kernel with array. 
+# Note: Base.dot product could also be used as input to `mapstencil`, but `kernelproduct`
```
Comment (rafaqz) [inline README.md:229]:
```suggestion
# Note: `Base.dot` product could also be used as input to `mapstencil`, but `kernelproduct` 
```

File: src/stencils/kernel.jl [hunk 4]
```diff
@@ -61,6 +61,33 @@ the stencil and kernel, like [`kernelproduct`](@ref) can be used in
...
+stencil = Window(1)
+
+# Create a stencil Kernel from the stencil and the kernel array
+k = Kernel(stencil, sharpen)
+
+# Wrap the random array and the Kernel in a StencilArray
+A = StencilArray(r, k)
+
+# use `mapstencil` with the `kernelproduct` function to convolve the Kernel with array
+# Note: Base.dot product could also be used as input to `mapstencil`, but `kernelproduct`
```
Comment (rafaqz) [inline src/stencils/kernel.jl:87]:
```suggestion
# Note: `Base.dot` product could also be used as input to `mapstencil`, but `kernelproduct` 
```

**Pull request merged**